asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py

```diff
@@ -60,6 +74,29 @@ def prep(
     def provenance(self) -> dict[str, str]:
         ...
 
+    @staticmethod
+    def cache(
+        prepped_complexes: list[PreppedComplex],
+        dir: Union[str, Path],
+        type=CacheType.DesignUnit,
+    ) -> None:
+        """
+        Cache a set of design units for use later.
+        """
+        dir = Path(dir)
+        if not dir.exists():
+            dir.mkdir(parents=True)
+
+        for pc in prepped_complexes:
+            if type == CacheType.DesignUnit:
+                du_name = pc.target.target_name + ".oedu"
+                du_path = dir / du_name
+                pc.target.to_oedu_file(du_path)
+            elif type == CacheType.JSON:
+                json_name = pc.target.target_name + ".json"
+                json_path = dir / json_name
+                pc.to_json_file(json_path)
+
 
 class ProteinPrepper(ProteinPrepperBase):
```

Comment:
I think we should stick to[ camel case](https://en.wikipedia.org/wiki/Camel_case) overall  for classes (which is general convention) and would leave us with choices between 

* SprucePrepper
* OEPrepper
* SomethingElse ? 

reply here and we can change it post hoc easily.

Reply:
How do you feel about `oeProteinPrepper` or `ProteinPrepperOE`? Again I think it makes sense to avoid classes with prefixed `OE` since that's what OpenEye uses for everything and could make it hard to distinguish between our code and OpenEye's? 
And since this prepper does more than spruce I think it makes sense to call it a protein prepper